File: backend/src/main/kotlin/outcobra/server/model/dto/ExamDto.kt
package outcobra.server.model.dto

import outcobra.server.model.Subject
import outcobra.server.model.interfaces.OutcobraDto
import outcobra.server.model.interfaces.ParentLink
import java.time.LocalDate

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
data class ExamDto(val id: Long = 0,
                   val name: String = "",
                   val description: String = "",
                   val date: LocalDate = LocalDate.now(),
                   val mark: MarkValueDto? = null,
                   var examTasks: MutableList<ExamTaskDto> = mutableListOf(),
                   val subject: SubjectDto = SubjectDto()) : OutcobraDto {

    override val identifier: Long
        get() = id
    override val parentLink: ParentLink
        get() = ParentLink.make(subject.id, Subject::class.java)
}

File: frontend/src/app/exam/model/exam.dto.ts
import { Moment } from 'moment';
import { Dto } from '../../core/common/dto';
import { SubjectDto } from '../../manage/model/manage.dto';
import { MarkDto } from '../../mark/model/mark.dto';
import { ExamTaskDto } from './exam.task.dto';

export interface ExamDto extends Dto {
  name: string
  description: string
  date: string | Moment
  mark: MarkDto
  examTasks: ExamTaskDto[]
  subject: SubjectDto
}
